File: Arrays/java/Inverse_of_an_Array.java
import java.io.*;
import java.util.*;

public class Main {
    public static void display(int[] a) {
        StringBuilder sb = new StringBuilder();

        for (int val: a) {
            sb.append(val + "
");
        }
        System.out.println(sb);
    }

    public static int[] inverse(int[] a) {
        // write your code here
        int n=a.length;
        int b[]=new int[n];
        for(int i=0;i<n;i++){
            b[a[i]]=i;
        }
        return b;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(br.readLine());
        }

        int[] inv = inverse(a);
        display(inv);
    }

}

File: 4. Arrays/cpp/Inverse-of-an-Array.cpp
#include <iostream>
using namespace std;

int *inverse(int *arr, int n)
{
    int *inv = new int[n];
    for (int i = 0; i < n; i++)
    {
        int pos = arr[i];
        inv[pos] = i;
    }
    return inv;
}

void display(int *arr, int n)
{
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << endl;
    }
    cout << endl;
}

int main()
{
    int n;
    cin >> n;
    int *arr = new int[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    int *inv = inverse(arr, n);
    display(inv, n);
}